staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go

```diff
@@ -292,28 +316,24 @@ func (o *ExposeServiceOptions) RunExpose(cmd *cobra.Command, args []string) erro
 			if err != nil {
 				return cmdutil.UsageErrorf(cmd, "couldn't find protocol via introspection: %v", err)
 			}
-			if protocols := generate.MakeProtocols(protocolsMap); !generate.IsZero(protocols) {
+			if protocols := MakeProtocols(protocolsMap); protocols != nil {
 				params["protocols"] = protocols
 			}
 		}
 
-		if generate.IsZero(params["labels"]) {
+		if params["labels"] == nil {
 			labels, err := meta.NewAccessor().Labels(info.Object)
 			if err != nil {
 				return err
 			}
 			params["labels"] = polymorphichelpers.MakeLabels(labels)
 		}
-		if err = generate.ValidateParams(names, params); err != nil {
-			return err
-		}
-		// Check for invalid flags used against the present generator.
-		if err := generate.EnsureFlagsValid(cmd, generators, generatorName); err != nil {
+		if err = ValidateParams(names, params); err != nil {
 			return err
 		}
 
 		// Generate new object
-		object, err := generator.Generate(params)
+		object, err := generateService(params)
```

Comment:
service, err := o.exposeService (I guess now it would be good actually to create a new function instead of using the old one, and porting the things to here)

Also, I was wondering if we wouldn't want to use pkg/cmd/create/create_service.go (and extend it) instead of re-implement everything here :)

One thing I'm thinking is if we can not check what flags the create service options miss, add to it, and instead of create a `ExposeServiceOptions` we could use the `create.ServiceOptions`, populate it, and call the command

@eddiezane need some adult opinion here :)